packages/gatsby/cache-dir/register-service-worker.js

```diff
@@ -23,8 +23,14 @@ if (
               if (navigator.serviceWorker.controller) {
                 // At this point, the old content will have been purged and the fresh content will
                 // have been added to the cache.
-                // We set a flag so Gatsby Link knows to refresh the page on next navigation attempt
-                window.GATSBY_SW_UPDATED = true
+                // We call the onServiceWorkerUpdateReady API so users can show update prompts.
+                apiRunner(`onServiceWorkerUpdateReady`, { serviceWorker: reg })
+
+                // If resources failed for the current page, reload.
+                if (window.___failedResources) {
+                  console.log(`resources failed, SW updated - reloading`)
+                  window.reload()
```

Comment:
shouldn't this be:
```suggestion
                  window.location.reload()
```
?

Reply:
oh yeah it should!

Reply:
yes, thanks for spotting! I made this same mistake a while ago with the docs for handling SW updates, oops :smile: